### PR TITLE
Multi linux arch and multi windows distro builds

### LINKED ISSUE #266: Add support for multi Windows distro builds
There are some implementation ideas from these two projects:

- [sig-storage's csi-release-tools](https://github.com/kubernetes-csi/csi-release-tools/issues/145)
- [gce persistent disk CSI driver' Dockerfile.Windows](https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/master/Dockerfile.Windows) - Uses csi-release-tools but it has its own way to build Windows images

/kind feature

### PR DESCRIPTION
**Pull request by mauriciopoppe**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
/kind feature

**What this PR does / why we need it**:

Adds support for multi windows distro builds, a summary of the changes:

- Moved from `docker build` to `docker buildx`
- Support for all the Windows distros (minus 2022)

I tested this with:

```
REGISTRY=gcr.io/mauriciopoppe-gke-dev ALLOW_UNSTABLE=true ALLOW_DIRTY=true VERSION=canary ./hack/release.sh
```

And commented parts of the code to push `latest` too, the output is:

image

Manifest contents:

```
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
   "manifests": [
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1160,
         "digest": "sha256:02b65ab8f73d9317e5ff84ee378551bee68a4e85948a795cad7689802a01401e",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1160,
         "digest": "sha256:aa0df19f3ca9b720e14d14e9b37fc9aaccbc7984dabb7567e3d757aa86f9dafd",
         "platform": {
            "architecture": "arm",
            "os": "linux",
            "variant": "v7"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1160,
         "digest": "sha256:3112838cb4f22608044ccbb1565472209f6a3c81a91ceb0db2aa804d19af21f5",
         "platform": {
            "architecture": "arm64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1160,
         "digest": "sha256:ba735b867e126750d15c0c4258ce5492f4945aa6c724e9e36e08bc50dfc624ed",
         "platform": {
            "architecture": "ppc64le",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1160,
         "digest": "sha256:9b195a6672192458f2290e90bcbca305c44fdad671b714ff7d2c62dbeb81395d",
         "platform": {
            "architecture": "s390x",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 956,
         "digest": "sha256:9b00b04c86a31af4e4ff17ba319675e845adf0b22ff4d299dae18105c55e8b31",
         "platform": {
            "architecture": "amd64",
            "os": "windows",
            "os.version": "10.0.18363.1556"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 957,
         "digest": "sha256:f1239689b5a9b55420ce1487bf38feb4cb538335654e884d43f17fac4c13e984",
         "platform": {
            "architecture": "amd64",
            "os": "windows",
            "os.version": "10.0.19041.1288"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 957,
         "digest": "sha256:682fbdb80ef2248502c81aa7ddd0f61e896473db87204b1ccd22970e08433b16",
         "platform": {
            "architecture": "amd64",
            "os": "windows",
            "os.version": "10.0.19042.1288"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 956,
         "digest": "sha256:7065d5e721771113f6fd6d69a6384cc6935749688ff11f2fcfe8a41178fc5fc4",
         "platform": {
            "architecture": "amd64",
            "os": "windows",
            "os.version": "10.0.17763.2237"
         }
      }
   ]
}
```

**Which issue(s) this PR fixes**:

Fixes #266

**Release note**:
```
Support for multi Windows distro builds
```

/sig storage
/cc @jingxu97 @cofyc

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -58,29 +60,46 @@ release:
...
+		--build-arg BUILDPLATFORM=linux \
+		--build-arg STAGINGVERSION=$(STAGINGVERSION) --push .
+
+build-and-push-container-windows-%: init-buildx
+	$(DOCKER) buildx build --file=./deployment/docker/Dockerfile.Windows --platform=windows/amd64 \
+		-t $(STAGINGIMAGE):$(STAGINGVERSION)_windows_$* \
+		--build-arg BASE_IMAGE=$(call lookup,$*,$(WINDOWS_DISTROS),$(WINDOWS_BASE_IMAGES)) \
+		--build-arg STAGINGVERSION=$(STAGINGVERSION) --push .
+
+test: build-container-linux-amd64
```
Comment (jingxu97) [inline Makefile:87]: the test is just for amd?

Comment (mauriciopoppe) [reply Makefile:87]: yes, the unit test only runs in linux amd64

**Pull request merged**